VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +122,110 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    public static class DiscussionResponsesLoader implements
+            InfiniteScrollUtils.PageLoader<DiscussionComment> {
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private Context context;
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private int nextPage = 1;
+        private String threadId;
+        private boolean isQuestionTypeThread;
+
+        private ThreadComments unendorsedComments;
+        private boolean isEndorsedFetched = false;
+
+        public DiscussionResponsesLoader(@NonNull Context context,@NonNull String threadId,
+                                         boolean isQuestionTypeThread) {
+            this.context = context;
+            this.threadId = threadId;
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        @Override
+        public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) {
+                getEndorsedList();
```

Comment:
> If there aren't enough endorsed comments to fill the screen, `loadNextPage` will get triggered a second time

Actually, from what I can see that won't happen in the current implementation, as described in the second section of this previous comment of mine here: https://github.com/edx/edx-app-android/pull/475#discussion_r47736012